RecoEgamma/EgammaTools/plugins/BuildFile.xml

```diff
@@ -3,6 +3,7 @@
   <use   name="FWCore/Framework"/>
   <use   name="PhysicsTools/MVAComputer"/>
   <use   name="PhysicsTools/MVATrainer"/>
+  <use   name="PhysicsTools/PatAlgos"/>
```

Comment:
no, reco:: objects use in ModifyObjectValueBase is not really required, if they are to be used outside of the PhysicsTools subsystem as well as in PhysicsTools, they just have to be defined in CommonTools, certainly not in the PatAlgos.
The other alternative is to move all code in RecoEgamma/EgammaTools that has to depend on PhysicsTools/PatAlgos to PatAlgos